set_2_block_crypto/challenge_9_implement_pkcs7_padding/pkcs7_pad.cpp

```diff
@@ -22,53 +22,43 @@ pkcs7_pad_streambuf::pkcs7_pad_streambuf(std::istream & inputStream,
 
 
 std::streambuf::pos_type pkcs7_pad_streambuf::seekpos(
-        std::streambuf::pos_type pos,
-        std::ios_base::openmode which
-)
+    std::streambuf::pos_type pos,
+    std::ios_base::openmode which)
 {
     assert(pos == std::streampos(0));
     assert(which == std::ios_base::in);
-    return pos == std::streampos(0)
-            && which == std::ios_base::in
-            && (d_inputStream->clear(), d_inputStream->seekg(pos))
-        ? (
-                delete [] d_paddingBuffer,
-                d_paddingBuffer = nullptr,
-                setg(nullptr, nullptr, nullptr),
-                d_inputLength = 0,
-                pos
-        )
-        : std::streambuf::seekpos(pos, which);
+    return pos == std::streampos(0) && which == std::ios_base::in
+                   && (d_inputStream->clear(), d_inputStream->seekg(pos))
+               ? (delete[] d_paddingBuffer,
+                  d_paddingBuffer = nullptr,
+                  setg(nullptr, nullptr, nullptr),
+                  d_inputLength = 0,
+                  pos)
+               : std::streambuf::seekpos(pos, which);
 }
 
 
 std::streambuf::pos_type pkcs7_pad_streambuf::seekoff(
-        std::streambuf::off_type off,
-        std::ios_base::seekdir dir,
-        std::ios_base::openmode which
-)
+    std::streambuf::off_type off,
+    std::ios_base::seekdir dir,
+    std::ios_base::openmode which)
 {
     assert(off == 0);
     assert(dir == std::ios_base::cur);
     assert(which == std::ios_base::in);
-    return off == 0
-            && dir == std::ios_base::cur
-            && which == std::ios_base::in
-        ? (
-                d_paddingBuffer
-                ? std::streampos(d_inputLength + gptr() - d_paddingBuffer)
-                : std::streampos(d_inputStream->tellg()
-                                 - std::streamoff(egptr() - gptr()))
-        )
-        : std::streambuf::seekoff(off, dir, which);
+    return off == 0 && dir == std::ios_base::cur && which == std::ios_base::in
+               ? (d_paddingBuffer
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested conditional operator into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3358)